network/src/commonTest/kotlin/com/wire/kalium/api/tools/json/api/user/client/ClientApiTest.kt

```diff
@@ -0,0 +1,87 @@
+package com.wire.kalium.api.tools.json.api.user.client
+
+import com.wire.kalium.api.ApiTest
+import com.wire.kalium.network.api.ErrorResponse
+import com.wire.kalium.network.api.prekey.PreKey
+import com.wire.kalium.network.api.user.client.Capabilities
+import com.wire.kalium.network.api.user.client.ClientApi
+import com.wire.kalium.network.api.user.client.ClientApiImp
+import com.wire.kalium.network.api.user.client.ClientType
+import com.wire.kalium.network.api.user.client.DeviceType
+import com.wire.kalium.network.api.user.client.LocationResponse
+import com.wire.kalium.network.api.user.client.RegisterClientRequest
+import com.wire.kalium.network.api.user.client.RegisterClientResponse
+import com.wire.kalium.network.tools.KtxSerializer
+import io.ktor.client.call.receive
+import io.ktor.client.features.ClientRequestException
+import io.ktor.http.HttpStatusCode
+import kotlinx.coroutines.test.runTest
+import kotlinx.serialization.encodeToString
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+
+class ClientApiTest : ApiTest {
+    @Test
+    fun givenAValidRegisterClientRequest_whenCallingTheRegisterClientEndpoint_theRequestShouldBeConfiguredCorrectly() =
+        runTest {
+            val httpClient = mockHttpClient(
+                KtxSerializer.json.encodeToString(VALID_Register_Client_RESPONSE),
+                statusCode = HttpStatusCode.Created,
+                assertion = {
+                    assertPost()
+                    assertJson()
+                    assertNoQueryParams()
+                    assertPathEqual(PATH_CLIENTS)
+                }
+            )
+            val clientApi: ClientApi = ClientApiImp(httpClient)
+            val response = clientApi.registerClient(Register_Client_REQUEST)
+            assertEquals(response.resultBody, VALID_Register_Client_RESPONSE)
+        }
+
+    @Test
+    fun givenAnInvalidLoginRequest_whenCallingTheLoginEndpoint_theCorrectExceptionIsThrown() = runTest {
```

Comment:
Same naming thing with the given condition. 
Maybe something like: `given an error response from the server` ?